src/fileimpl.cpp

```diff
@@ -233,9 +234,12 @@ class Grouping
     quickCheckForCorruptFile();
 
     mp_titleDirentAccessor = getTitleAccessor("listing/titleOrdered/v1");
+    if (header.getMinorVersion() >= Fileheader::zimMinorVersionWithoutV0) {
+      const_cast<bool&>(m_hasExhaustiveEntryTitleListing) = false;
```

Comment:
Another instance of this `const_cast`-of-a-`const`-data-member hack is ugly. Luckily it is not justified in this case. You can have a private `bool hasExhaustiveEntryTitleListing() const` (or, better, `hasTitleListingV0()`) instead. And, frankly, I don't think that we will notice any performance degradation if we get rid of `m_newNamespaceScheme` in the same manner.

Reply:
Indeed. I have move all in methods in `FileHeader`.